flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraph.java

```diff
@@ -335,9 +352,16 @@ public static ArchivedExecutionGraph createFrom(ExecutionGraph executionGraph) {
 
         final long[] timestamps = new long[JobStatus.values().length];
 
+        // if the state is overridden with a non-globally-terminal state then we need to erase
+        // traces of globally-terminal states
+        final boolean clearGloballyTerminalStateTimestamps =
+                statusOverride != null && !statusOverride.isGloballyTerminalState();
```

Comment:
I think we check in the beginning of this method whether `statusOverride` is not a globally terminal state if it is not `null`.